CRM/Contribute/BAO/Contribution.php

```diff
@@ -4177,8 +4177,14 @@ private static function getOriginalContribution($contributionID) {
    *
    * @throws \CRM_Core_Exception
    */
-  public static function updateMembershipBasedOnCompletionOfContribution($contributionID, $changeDate) {
+  public static function updateMembershipBasedOnCompletionOfContribution(int $contributionID, string $changeDate) {
+    CRM_Core_Error::deprecatedWarning('updateMembershipBasedOnCompletionOfContribution has been moved to \Civi\Membership\OrderCompleteSubscriber and will be removed from CRM_Contribute_BAO_Contribution');
+
     $memberships = self::getRelatedMemberships((int) $contributionID);
+    if (empty($memberships)) {
```

Comment:
Why did this get added if the function is presumably no longer called?

Reply:
Right, I added it before copying the function over. It's not needed here (in fact not needed at all but helps with making the code clear). I'll remove it from here.

Reply:
Done